rebasehelper/tests/test_specfile.py

```diff
@@ -199,6 +199,36 @@ def test_find_archive_target_in_prep(self, spec_object):
         target = spec_object.find_archive_target_in_prep('misc.zip')
         assert target == 'test-1.0.2/misc'
 
+    def test_write_updated_patches(self, mock_spec_object_generator):
+        content = """Patch0:    0.patch
+Patch1:    1.patch
+
+%patchlist
+2.patch
+3.patch
+
+%prep
+%patch0 -p0
+%patch1 -p1
+%patch2 -p0
+%patch3 -p0
+"""
+        expected_content = """Patch0:    0.patch
+
+%patchlist
+2.patch
+#3.patch
+
+%prep
+%patch0 -p0
+%patch1 -p0
+# The following patch contains conflicts
+#%%patch3 -p0
+"""
+        spec = mock_spec_object_generator({'save': lambda _: None}, {'spec_content': content, 'removed_patches': []})
+        spec.write_updated_patches({'deleted': ['1.patch'], 'inapplicable': ['3.patch']}, True)
```

Comment:
And now swap deleted with inapplicable and watch the test fail.

Reply:
It seems to me that even after swapping, the code for modifying prep works correctly. Here is the spec content after swapping inapplicable and deleted.

```
Patch0:    0.patch
#Patch1:   1.patch

%patchlist
2.patch

%prep
%patch0 -p0
# The following patch contains conflicts
#%%patch1 -p1
%patch1 -p0
```

So 3.patch was removed from patchlist, it was removed from prep. 1.patch was commented out and so the index of 2.patch changed to 1 and it is reflected in prep. Am I missing something?

Reply:
Sorry, my bad. But what if you set `disable_inapplicable` to `False`?

Reply:
Good catch, there was indeed a bug. I think some more tests for this method should be added to test various possibilities (at least test both states of `disable_inapplicable`). But I'd ideally want to avoid having long strings in the tests, I think they decrease readability quite a lot (those 2 10-line strings are not ideal). Do you have any ideas how we could do it in a better way?

Reply:
Other than dividing the tests into more individual files? No, I don't think there is a way.